From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rita Wolf <ritasistr@mail.ru>
Date: Thu, 29 May 2025 19:25:27 +0300
Subject: [PATCH] fix crash and added checking isTickThreadFor for exclude data
 race


diff --git a/net/minecraft/world/entity/ai/village/poi/PoiManager.java b/net/minecraft/world/entity/ai/village/poi/PoiManager.java
index c10810bf00d75f459c3c6a9415c1e09f0519d50e..766e77889bb1941eff76039476cf4fb5978a7686 100644
--- a/net/minecraft/world/entity/ai/village/poi/PoiManager.java
+++ b/net/minecraft/world/entity/ai/village/poi/PoiManager.java
@@ -212,7 +212,7 @@ public class PoiManager extends SectionStorage<PoiSection, PoiSection.Packed> im
 
     public Stream<PoiRecord> getInSquare(Predicate<Holder<PoiType>> typePredicate, BlockPos pos, int distance, PoiManager.Occupancy status) {
         int i = Math.floorDiv(distance, 16) + 1;
-        return ChunkPos.rangeClosed(new ChunkPos(pos), i).flatMap(chunkPos -> this.getInChunk(typePredicate, chunkPos, status)).filter(poiRecord -> {
+        return ChunkPos.rangeClosed(new ChunkPos(pos), i).filter(cpos -> ca.spottedleaf.moonrise.common.util.TickThread.isTickThreadFor(this.world, cpos)).flatMap(chunkPos -> this.getInChunk(typePredicate, chunkPos, status)).filter(poiRecord -> {
             BlockPos pos1 = poiRecord.getPos();
             return Math.abs(pos1.getX() - pos.getX()) <= distance && Math.abs(pos1.getZ() - pos.getZ()) <= distance;
         });
